internal/service/streamprocessor/state_transition.go

```diff
@@ -0,0 +1,61 @@
+package streamprocessor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
+)
+
+const (
+	InitiatingState = "INIT"
+	CreatingState   = "CREATING"
+	CreatedState    = "CREATED"
+	StartedState    = "STARTED"
+	StoppedState    = "STOPPED"
+	DroppedState    = "DROPPED"
+	FailedState     = "FAILED"
+)
+
+func WaitStateTransition(ctx context.Context, requestParams *admin.GetStreamProcessorApiParams, client admin.StreamsApi, pendingStates, desiredStates []string) (*admin.StreamsProcessorWithStats, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending:    pendingStates,
+		Target:     desiredStates,
+		Refresh:    refreshFunc(ctx, requestParams, client),
+		Timeout:    1 * time.Minute,
+		MinTimeout: 3 * time.Second,
```

Comment:
curious why these 3 seconds are needed

Reply:
Usually state transitions for stream processors are very quick, but I left the 3 seconds to be safe so that the smallest time to wait before refreshes would be enough for the transition to be complete. Let me know if you think this should be changed.